x-pack/plugins/ml/server/models/data_frame_analytics/validation.ts

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { IScopedClusterClient } from 'kibana/server';
+import { getAnalysisType } from '../../../common/util/analytics_utils';
+import {
+  INCLUDED_FIELDS_THRESHOLD,
+  MINIMUM_NUM_FIELD_FOR_CHECK,
+  FRACTION_EMPTY_LIMIT,
+  TRAINING_DOCS_LOWER,
+  TRAINING_DOCS_UPPER,
+  VALIDATION_STATUS,
+} from '../../../common/constants/validation';
+import { getDependentVar } from '../../../common/util/analytics_utils';
+import { extractErrorMessage } from '../../../common/util/errors';
+import { SearchResponse7 } from '../../../common';
+import { DataFrameAnalyticsConfig } from '../../../common/types/data_frame_analytics';
+
+interface MissingAgg {
+  [key: string]: {
+    doc_count: number;
+  };
+}
+interface CardinalityAgg {
+  [key: string]: {
+    value: number;
+  };
+}
+
+type ValidationSearchResult = Omit<SearchResponse7, 'aggregations'> & {
+  aggregations: MissingAgg | CardinalityAgg;
+};
+
+const defaultQuery = { match_all: {} };
+
+const trainingPercentHeading = i18n.translate(
+  'xpack.ml.models.dfaValidation.messages.trainingPercentHeading',
+  {
+    defaultMessage: 'Training percent',
+  }
+);
+const analysisFieldsHeading = i18n.translate(
+  'xpack.ml.models.dfaValidation.messages.analysisFieldsHeading',
+  {
+    defaultMessage: 'Analysis fields',
+  }
+);
+const dependentVarHeading = i18n.translate(
+  'xpack.ml.models.dfaValidation.messages.dependentVarHeading',
+  {
+    defaultMessage: 'Dependent variable',
+  }
+);
+const dependentVarWarningMessage = {
+  id: 'dep_var_check',
+  text: '',
+  status: VALIDATION_STATUS.WARNING,
+  heading: dependentVarHeading,
+};
+const analysisFieldsWarningMessage = {
+  id: 'analysis_fields',
+  text: '',
+  status: VALIDATION_STATUS.WARNING,
+  heading: analysisFieldsHeading,
+};
+
+function getTrainingPercentMessage(trainingDocs: number) {
+  let trainingPercentMessage;
```

Comment:
`trainingPercentMessage` probably isn't needed as we could return from within the `if` statement below.

Reply:
Updated in a7e5fc69c41f0b104667516fc53457eebf0a649d